viadot/sources/s3.py

```diff
@@ -1,266 +1,132 @@
-    ):
-        if quoting is None:
-            quoting = 0
-
-        path = path or self.path
-        url = os.path.join(self.base_url, path)
+        return wr.s3.does_object_exist(boto3_session=self.session, path=path)
 
-        if url.endswith(".csv"):
-            df = pd.read_csv(
-                url,
-                storage_options=self.storage_options,
-                sep=sep,
-                quoting=quoting,
-                lineterminator=lineterminator,
-                error_bad_lines=error_bad_lines,
-            )
-        elif url.endswith(".parquet"):
-            df = pd.read_parquet(url, storage_options=self.storage_options)
-        else:
-            raise ValueError("Only CSV and parquet formats are supported.")
-
-        return df
-
-    def ls(self, path: str = None) -> List[str]:
+    def cp(self, paths: List[str], from_path: str, to_path: str):
         """
-        Returns a list of files in a path.
+        Copies the contents of `from_path` to `to_path`.
 
         Args:
-            path (str, optional): Path to a folder. Defaults to None.
+            paths (List[str]): List of S3 objects paths (e.g. [s3://bucket/dir0/key0, s3://bucket/dir0/key1]).
+            from_path (str, optional): S3 Path for the source directory.
+            to_path (str, optional): S3 Path for the target directory.
         """
-        path = path or self.path
-        return self.fs.ls(path)
 
-    def rm(self, path: str = None, recursive: bool = False):
+        wr.s3.copy_objects(
+            boto3_session=self.session,
+            paths=paths,
+            source_path=from_path,
+            target_path=to_path,
+        )
+
+    def rm(self, path: str):
         """
         Deletes files in a path.
-
         Args:
-            path (str, optional): Path to a folder. Defaults to None.
-            recursive (bool, optional): Whether to delete files recursively.
-                Defaults to False.
+            path (str): Path to a folder.
         """
-        path = path or self.path
-        self.fs.rm(path, recursive=recursive)
 
-    def cp(self, from_path: str = None, to_path: str = None, recursive: bool = False):
-        """
-        Copies the contents of `from_path` to `to_path`.
-
-        Args:
-            from_path (str, optional): Path from which to copy file(s).
-                Defauls to None.
-            to_path (str, optional): Path where to copy file(s). Defaults
-                to None.
-            recursive (bool, optional): Whether to copy file(s) recursively.
-                Defaults to False.
-        """
-        from_path = from_path or self.path
-        to_path = to_path
-        self.fs.cp(from_path, to_path, recursive=recursive)
+        wr.s3.delete_objects(boto3_session=self.session, path=path)
 
     def from_df(
-        self, df: pd.DataFrame, path: str = None, overwrite: bool = False
-    ) -> None:
+        self,
+        df: pd.DataFrame = None,
+        to_path: str = None,
```

Comment:
I don't think the "to_" prefix is required here. It's used in the "copy" method because there we have two path args that need to be distinguished.

Reply:
Fixed as per [🎨 fixed variable names](https://github.com/dyvenia/viadot/pull/579/commits/535d222edec696e411e2bd1d9d50e6a584a492e6)